effect/test_intents.py

```diff
@@ -46,3 +58,71 @@ def test_perform_func(self):
             TypeDispatcher({Func: perform_func}),
             Effect(intent))
         self.assertEqual(result, "foo")
+
+
+class ParallelTests(TestCase):
+    """Tests for :func:`parallel`."""
+
+    def test_empty(self):
+        """
+        When given an empty list of effects, ``perform_parallel_async`` returns
+        an empty list synchronusly.
+        """
+        result = sync_perform(
+            TypeDispatcher({ParallelEffects: perform_parallel_async}),
+            parallel([]))
+        self.assertEqual(result, [])
+
+    def test_parallel(self):
+        """
+        'parallel' results in a list of results of the given effects, in the
+        same order that they were passed to parallel.
+        """
+        result = sync_perform(
+            TypeDispatcher({ParallelEffects: perform_parallel_async,
+                            Constant: perform_constant}),
+            parallel([Effect(Constant('a')),
+                      Effect(Constant('b'))]))
+        self.assertEqual(result, ['a', 'b'])
+
+    def test_error(self):
+        """
+        When given an effect that results in a Error,
+        ``perform_parallel_async`` result in ``FirstError``.
+        """
+        self.assertThat(
+            lambda: sync_perform(
+                TypeDispatcher({ParallelEffects: perform_parallel_async,
+                                Error: perform_error}),
+                parallel([Effect(Error(EquibleException(message="foo")))])),
+            raises(FirstError(exception=EquibleException(message='foo'),
+                              index=0)))
```

Comment:
Also, need to test that the index accurately reflects where the error came from.